### PR TITLE
fix(publication): source A0-A11 sequence from live tracker

### PR DESCRIPTION
**Pull request by joeharris76**

### CODE REVIEW

File: scripts/publication/check_plan_reconciliation.py [hunk 1]
```diff
@@ -53,16 +77,25 @@ def main() -> int:
...
-    expected_ids = [item_id for item_id in REQUIRED_TRACKER_IDS if item_id.startswith(args.todo_prefix)]
-    if tracker_ids != expected_ids:
-        missing.append("exact ordered A0-A11 tracker sequence")
+    live_items = load_live_items()
+
+    if live_items is None:
+        print(
+            "WARNING: live tracker unavailable; skipping live A0-A11 reconciliation (advisory mode)",
+            file=sys.stderr,
+        )
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py:244]: Addressed in 1d2dad722: the reconciliation gate is now fail-closed. When `load_live_items()` returns None (no credential, wrapper timeout/failure, or non-list/invalid payload), `main()` appends the 'live tracker unavailable' surface to `missing` and returns 1 (hard failure) instead of passing in advisory mode. A plan can no longer reconcile solely on document-only checks; success requires an actual live tracker read that matches the currently pending migration order. The docstring now states the fail-closed contract. Coverage added: `test_main_fails_closed_when_live_tracker_unavailable` (None -> exit 1), `test_main_fails_when_live_sequence_does_not_match` (stale/partial live sequence -> exit 1), and `test_main_succeeds_when_live_sequence_matches_decision` (matching live payload -> exit 0). Focused suite: 13 passed; ruff clean; pre-push pr-preflight passed.

**Comment by joeharris76** [comment]
@codex review

File: scripts/publication/check_plan_reconciliation.py [hunk 2]
```diff
@@ -25,26 +36,40 @@
...
 
 
+def _phase_key(item_id: str, prefix: str) -> tuple[int, str]:
+    match = re.search(re.escape(prefix) + r"a(\d+)-", item_id)
+    return (int(match.group(1)) if match else 0, item_id)
+
+
+def live_tracker_ids(items: list[dict], prefix: str) -> list[str]:
+    """Order tracker items matching ``prefix`` by their numeric A-phase (a0..a11)."""
+    matches = [item["id"] for item in items if str(item.get("id", "")).startswith(prefix)]
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py]: Addressed in c65991430: `live_tracker_ids()` now excludes dropped items. `todo list` returns items across all lifecycle states, so a step moved to `dropped` still appeared in the output and was counted in the live sequence, letting the gate pass while a migration step had been dropped. The comprehension now filters out any item whose `state` is `dropped` (case-insensitive `str(item.get('state','')).lower() != 'dropped'`), so the live sequence reflects only the currently pending migration order. The docstring states the behavior. Coverage added: `test_live_tracker_ids_excludes_dropped_items` (done/active retained, dropped excluded). Focused suite: 14 passed; ruff clean; pre-push pr-preflight passed.

**Comment by joeharris76** [comment]
@codex review

File: scripts/publication/check_plan_reconciliation.py [hunk 3]
```diff
@@ -25,26 +36,50 @@
...
+    ``dropped`` still appears in output and would otherwise be counted as part
+    of the live sequence. Exclude dropped items so the reconciliation cannot
+    pass while a step of the migration has been dropped.
+    """
+    matches = [
+        item["id"]
+        for item in items
+        if str(item.get("id", "")).startswith(prefix) and str(item.get("state", "")).lower() != "dropped"
+    ]
+    return sorted(matches, key=lambda item_id: _phase_key(item_id, prefix))
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py]: Addressed in 053ff4efd: the live sequence is now read from the tracker dependency graph, not inferred from the A-phase in the ID.

- `live_tracker_ids()` now returns a topological order of the non-dropped matching items derived from each row's `deps` list (Kahn's algorithm, deterministic phase-key tie-break among parallel items), and raises `ValueError` on a dependency cycle.
- Because `todo list --json` does not expose dependencies, dependencies are read with `todo show <id> --json` via a new `load_live_deps()`; if any dependency read fails the gate fails closed (consistent with the unavailable-tracker handling).
- `main()` attaches each item's deps before ordering and treats an unreadable dependency graph or a cycle as a hard failure. Dropped items remain excluded.

Sanity check: the live DAG (e.g. a6/a7 both depend on a5; a8 depends on a2,a4,a7; a9 depends on a6,a7,a8) topologically orders to exactly a0..a11, matching the decision, so the gate does not false-fail on current deps; a deps renumbering that diverges from the names now fails instead of silently passing. Coverage added/updated: `test_live_tracker_ids_orders_by_dependency_graph_not_id_number` (deps contradict numeric order -> dependency order wins), `test_live_tracker_ids_detects_dependency_cycle`, `test_load_live_deps_*`, `test_main_fails_closed_when_dependency_graph_unavailable`, and the existing main/tracker tests updated to inject deps. Focused suite: 19 passed; ruff clean; pre-push pr-preflight passed.

**Comment by joeharris76** [comment]
@codex review

File: scripts/publication/check_plan_reconciliation.py [hunk 4]
```diff
@@ -25,26 +36,94 @@
...
+    deps_of = {
+        item_id: [dep for dep in (item.get("deps") or []) if dep in selected] for item_id, item in selected.items()
+    }
+    order: list[str] = []
+    remaining = set(selected)
+    while remaining:
+        ready = sorted(
+            (item_id for item_id in remaining if not (set(deps_of[item_id]) & remaining)),
+            key=lambda item_id: _phase_key(item_id, prefix),
+        )
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py:123]: Addressed in 280b06072: validates dependency-edge drift directly instead of inferring order from the A-phase tie-break.

Contract (user choice: "Validate all implied precedence edges"): the plan's cited total order is authoritative. `dependency_violations(plan_order, deps)` fails on:
- **backward edge** — `"<item> depends on <dep>, which the plan orders after it"` (added/renumbered edge to a later item; cycles surface as this)
- **removed chain** — `"<item> has no prior dependency in the plan (removed chain)"` for rank>0 with no earlier in-set dep (e.g. A1 deps=[] while still ordered after A0)

`main()` now:
- gates set-equality on `sorted(live_ids, key=_phase_key) == tracker_ids` (missing)
- plus `dependency_violations(tracker_ids, live_deps)` — no longer compares a lossy topological order
- live deps sourced via `load_live_deps()` = `show <id> --json`; fail-closed if graph unavailable

Real-DAG sanity (verified `load_live_deps` input, not `todo list`):
`REAL_DEPS` a0..a11 (a6,a7→a5; a8→a2,a4,a7; a9→a6,a7,a8; a10→a9; a11→a10) → `dependency_violations(LIVE_A0_A11, REAL_DEPS) == []` (also `uv run python` prints `[]`).

Tests (`tests/unit/scripts/publication/test_plan_reconciliation.py`, 24 passed, ruff clean):
- `test_dependency_violations_passes_for_real_dag`
- `test_dependency_violations_fails_on_removed_chain` (a1 deps=[] → orphan)
- `test_dependency_violations_fails_on_backward_edge` (a1→a2 → backward)
- `test_dependency_violations_surfaces_cycle_as_backward_edge` (a0→a1 cycle)
- `test_main_succeeds_when_live_sequence_matches_decision` now injects `REAL_DEPS` (empty-deps would correctly fail as orphans)
- `test_main_fails_on_dependency_edge_drift` (all 12 present, a1→[] → 1)

`live_tracker_ids()` kept as tested topological helper (used by unit tests); `main()` no longer relies on its tie-break, so the stale-decision pass is eliminated while the legitimate parallel DAG (a6/a7 parallel under a5) remains valid.

**Comment by joeharris76** [comment]
@codex review

File: scripts/publication/check_plan_reconciliation.py [hunk 5]
```diff
@@ -25,26 +36,121 @@
...
+    rank = {item_id: i for i, item_id in enumerate(plan_order)}
+    order_set = set(plan_order)
+    violations: list[str] = []
+    for item_id in plan_order:
+        item_deps = [dep for dep in deps.get(item_id, []) if dep in order_set]
+        for dep in item_deps:
+            if rank[dep] >= rank[item_id]:
+                violations.append(f"{item_id} depends on {dep}, which the plan orders after it")
+        if rank[item_id] > 0 and not any(rank[dep] < rank[item_id] for dep in item_deps):
+            violations.append(f"{item_id} has no prior dependency in the plan (removed chain)")
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py:172]: Addressed in 238c46fa2: enforces exact dependency edge set, closing the "last prior edge" gap.

Previous `dependency_violations` only flagged orphan (no prior dependency) and backward edges, so removing a single required edge while retaining another earlier edge left a valid topological order and passed (e.g. A8 depends on A2,A4,A7 → live A8→A2,A4 still has prior deps, so no orphan, no backward).

Fix:
- Added `EXPECTED_DEPS` (freeze-time graph a0..a11, 12 items: a0→[], a1→a0, a2→a1, a3→a1,a2, a4→a3, a5→a4, a6→a5, a7→a5, a8→a2,a4,a7, a9→a6,a7,a8, a10→a9, a11→a10) as source of truth.
- `dependency_violations(plan_order, deps)` now, for each item in `EXPECTED_DEPS`, also reports:
  - `"<item> is missing expected dependency on <dep>"` for each expected in-set edge absent in live
  - `"<item> has unexpected dependency on <dep>"` for each live in-set edge not in expected
  - plus existing backward (`depends on ... which the plan orders after it`) and orphan checks. Cycle still surfaces as backward; fail-closed on graph unavailable remains.

Verification:
- `dependency_violations(LIVE_A0_A11, REAL_DEPS) == []` (real DAG still passes)
- Partial-removal drift now fails: A8→A2,A4 (missing A7) → `["a8 ... is missing expected dependency on ...a7-explorer..."]` (previously [])
- Unexpected extra edge fails: A3→A1,A2,A0 (extra A0) → `["a3 ... has unexpected dependency on ...a0..."]`
- `test_expected_deps_matches_real_deps` ensures test fixture matches script constant.

Tests (`tests/unit/scripts/publication/test_plan_reconciliation.py`, 28 passed, ruff clean):
- `test_dependency_violations_passes_for_real_dag`
- `test_dependency_violations_fails_on_removed_chain` (a1→[])
- `test_dependency_violations_fails_on_backward_edge` (a1→a2)
- `test_dependency_violations_surfaces_cycle_as_backward_edge`
- `test_dependency_violations_fails_on_partial_edge_removal` (A8 missing A7)
- `test_dependency_violations_fails_on_unexpected_edge`
- `test_main_succeeds_when_live_sequence_matches_decision` (injects REAL_DEPS)
- `test_main_fails_on_dependency_edge_drift` and `test_main_fails_on_partial_edge_removal` (both 1)

`live_tracker_ids()` kept as topological helper; `main()` gates on `sorted(live_ids)==tracker_ids` plus `dependency_violations(tracker_ids, live_deps)` with exact edge comparison, so any edge addition/removal now fails without a decision update, while legitimate parallel DAG (a6/a7→a5) remains valid because it matches EXPECTED_DEPS.

**Comment by joeharris76** [comment]
@codex review

**Comment by joeharris76** [comment]
@codex review

File: scripts/publication/check_plan_reconciliation.py [hunk 6]
```diff
@@ -53,16 +215,40 @@ def main() -> int:
...
+        print(
+            "ERROR: live tracker unavailable; cannot reconcile the A0-A11 sequence (fail closed)",
+            file=sys.stderr,
+        )
+        missing.append("exact ordered A0-A11 tracker sequence (live tracker unavailable)")
+    else:
+        live_scope = [
+            item
+            for item in live_items
+            if str(item.get("id", "")).startswith(args.todo_prefix) and str(item.get("state", "")).lower() != "dropped"
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py]: Addressed in 9a389c3: retains dropped required phases as explicit drift, so a dropped phase cannot be hidden by editing the decision to omit it.

Previous `main()` filtered `state == dropped` before any comparison:
`live_scope = [i for i in live_items if ... != dropped]` → `live_ids` (11) vs `tracker_ids` (11, if decision edited to omit) → match, and `dependency_violations` filtered via `order_set` (11) so references to the omitted ID were also ignored → gate passed with 11 phases.

Fix:
- Before filtering, builds `prefix_all` (all live items with prefix, including dropped).
- `dropped_required` = dropped where `id in EXPECTED_DEPS` (12 required) plus dropped where `id in tracker_ids` (forward-compat).
- For each `did` in sorted order, appends `dropped required phase is dropped: <did>` to `missing` (fail-closed). This is checked regardless of set equality, so 11-vs-11 still fails.
- Then builds `live_scope` filtered as before for set-equality and `dependency_violations(tracker_ids, live_deps)` (now with exact edge set: missing/unexpected plus backward/orphan). Dropped case therefore produces at least the dropped message plus orphan/missing for downstream dependents (e.g. a6/a7 lose a5).

Verification:
- `live_items = 12 × done` + `REAL_DEPS` → `main()==0` (real DAG passes)
- `live = 12 with a5 dropped, tracker=12` → `main()==1` via `dropped required phase is dropped: a5` + `exact ordered ... (live tracker)` + orphan/missing for a5
- `live = 11 actives + a5 dropped, tracker = 11 (omit a5)` → `main()==1` via `dropped required phase is dropped: a5` plus `a6/a7 has no prior dependency` (previously would have passed with 0). Both cases now correctly fail.

Tests (`tests/unit/scripts/publication/test_plan_reconciliation.py`, 30 passed, ruff clean):
- `test_main_fails_when_required_phase_is_dropped` (a5 dropped, tracker 12)
- `test_main_fails_when_dropped_phase_omitted_from_decision` (a5 dropped, tracker 11 via patched `planned_tracker_ids`)
- existing 28 tests still pass (including partial-removal and unexpected-edge cases for exact edge set, and dropped filtering for `live_tracker_ids` helper remains: `test_live_tracker_ids_excludes_dropped_items`).

`EXPECTED_DEPS` (12) remains source of truth for edge drift; `live_tracker_ids` still excludes dropped for its own ordering tests, but `main()` now explicitly retains dropped evidence to fail.

**Comment by joeharris76** [comment]
@codex review

File: scripts/publication/check_plan_reconciliation.py [hunk 7]
```diff
@@ -25,26 +36,177 @@
...
+    edge" gap where only orphan detection would fire.
+
+    ``deps`` maps each plan item to its live dependency ids (only those within
+    the plan set are considered).
+    """
+    rank = {item_id: i for i, item_id in enumerate(plan_order)}
+    order_set = set(plan_order)
+    violations: list[str] = []
+    for item_id in plan_order:
+        item_deps = [dep for dep in deps.get(item_id, []) if dep in order_set]
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py]: Addressed in cae8f63f: validates dependencies outside the A0–A11 set as unexpected external edges.

Previous `dependency_violations` filtered live deps to `order_set` before any check:

`item_deps = [dep for dep in deps.get(item_id, []) if dep in order_set]` → external deps were discarded before ordering, missing/unexpected, and orphan checks, so an A-phase gaining a dependency on any tracker item outside A0–A11 while all in-set edges matched would still return `[]` and the gate would pass, even though tracker readiness requires every `deps` predecessor to be done.

Fix:
- Now iterates over `live_all = deps.get(item_id, [])` first, and for each `dep not in order_set` appends `"<item> has unexpected external dependency on <dep>"` before the in-set logic. This ensures any external edge is explicit drift; there is no pinned external allowlist, so any such edge fails. In-set logic then uses `item_deps = [dep for dep in live_all if dep in order_set]` for backward, orphan, missing-expected and unexpected-in-set checks (which already compare against `EXPECTED_DEPS`).

Verification:
- `LIVE_A0_A11` with `REAL_DEPS` → `[]` (no external, still passes)
- Drift: `a3 → a1,a2,external-item` → `["a3 ... has unexpected external dependency on external-item"]` (previously `[]`)
- Existing in-set drift cases still fail as before (partial-removal `a8 missing a7`, backward `a1→a2`, orphan `a1→[]`, unexpected in-set extra).

Tests: `test_dependency_violations_fails_on_external_dependency` added; 33 passed total, ruff clean. The fix complements the exact-edge set enforcement (EXPECTED_DEPS) so both in-set and out-of-set drift are now caught.

File: scripts/publication/check_plan_reconciliation.py [hunk 8]
```diff
@@ -25,26 +36,177 @@
...
+    deps: dict[str, list[str]] = {}
+    for item_id in item_ids:
+        try:
+            result = subprocess.run(
+                [*shim_cmd, "show", item_id, "--json"], capture_output=True, text=True, check=True, timeout=60
+            )
+            item = json.loads(result.stdout)
+        except (OSError, subprocess.CalledProcessError, subprocess.TimeoutExpired, json.JSONDecodeError):
+            return None
+        deps[item_id] = item.get("deps") or []
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py:210]: Addressed in cae8f63f: rechecks lifecycle state after loading dependencies, closing the list→show race.

Previous `load_live_deps` did:

`item = json.loads(result.stdout); deps[item_id] = item.get("deps") or []`

discarding `state` from the `show --json` response. When a required phase is dropped after the initial `list --json` (which reported `state != dropped` and was used to build `live_ids`) but before its `show --json`, the fresher `show` payload carries `state: dropped`, but the old code ignored it and the gate could return success using the stale non-dropped state from the list, with `dependency_violations` also filtering via `order_set`.

Fix:
- `load_live_deps` now checks `if str(item.get("state","")).lower() == "dropped": return None` immediately after parsing each `show` response. Returning `None` triggers the existing fail-closed path in `main()` (`"dependency graph unavailable"` → `missing.append(...)` → `return 1`), so the gate fails rather than certifying the stale list.

Verification:
- Mock `show` returning `{"id": "a5", "state": "dropped", "deps": []}` → `load_live_deps([a5], cmd) is None` (previously `{"a5": []}`).
- `main()` with `live_items = 12×done` but `load_live_deps` simulating dropped race (`return None` for a5) → `1` (previously would have passed if deps matched).
- Existing dropped detection (list-based `dropped required phase is dropped: ...` for items that were already dropped at list time) remains, so both the race and the steady-state dropped cases are now covered. The gate still fails closed when the tracker is unavailable or the graph cannot be read.

Tests: `test_load_live_deps_returns_none_when_show_is_dropped` and `test_main_fails_when_dependency_show_reports_dropped` added; 33 passed, ruff clean. The fix preserves the explicit dropped-phase check in `main()` and adds the fresher-state check in `load_live_deps`.

**Comment by joeharris76** [comment]
@codex review

File: scripts/publication/check_plan_reconciliation.py [hunk 9]
```diff
@@ -25,26 +36,195 @@
...
+        for dep in live_all:
+            if dep not in order_set:
+                violations.append(f"{item_id} has unexpected external dependency on {dep}")
+        item_deps = [dep for dep in live_all if dep in order_set]
+        for dep in item_deps:
+            if rank[dep] >= rank[item_id]:
+                violations.append(f"{item_id} depends on {dep}, which the plan orders after it")
+        if rank[item_id] > 0 and not any(rank[dep] < rank[item_id] for dep in item_deps):
+            violations.append(f"{item_id} has no prior dependency in the plan (removed chain)")
+        if item_id in EXPECTED_DEPS:
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py:173]: Addressed in 3c254a2: rejects unpinned phases missing from the dependency pin.

Previous conditional `if item_id in EXPECTED_DEPS:` skipped exact-edge validation when a new ID was absent from the map. So when live tracker and decision both added a new matching phase (e.g. A12), `tracker_ids == live_item_ids` (13 vs 13) succeeded, and `dependency_violations` only ran backward/orphan checks for that ID (since `item_deps` filtered to `order_set` would be considered topo-valid). Any earlier dependency for the new phase would therefore pass reconciliation even though the graph is unpinned/incomplete.

Fix in `dependency_violations`:
- After the `if item_id in EXPECTED_DEPS:` block that checks missing/ unexpected in-set edges, added `else:` that appends `"<item> is not pinned in EXPECTED_DEPS (unpinned phase)"`. This makes every planned ID require a pin; existing 12 (`a0..a11`) remain pinned and still pass with `REAL_DEPS`, while any new phase (e.g. `a12-new-phase → a11`) now fails until `EXPECTED_DEPS` is updated to include it.

Verification:
- `dependency_violations(LIVE_A0_A11, REAL_DEPS) == []` (still passes)
- `plan = LIVE_A0_A11 + ["a12-new-phase"], deps[a12]=[a11]` → `["a12-new-phase is not pinned in EXPECTED_DEPS (unpinned phase)"]` (previously `[]`)

Tests: `test_dependency_violations_fails_on_unpinned_phase` added; 34 passed total, ruff clean. The fix ensures the gate fails closed on unpinned phases and complements the existing dropped/external/ exact-edge drift detection.

**Comment by joeharris76** [comment]
@codex review

File: scripts/publication/check_plan_reconciliation.py [hunk 10]
```diff
@@ -25,26 +36,197 @@
...
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py:173]: Addressed in 39ed96d: requires every pinned phase to remain present in both plan and live.

Previous `main()` only compared `tracker_ids vs live_item_ids` and `dependency_violations` only visited plan IDs (to flag unpinned via `else:`). When a pinned terminal phase (e.g. A11) disappeared from both tracker and decision (both 11 after restore/import), `tracker_ids == live_item_ids` (11 vs 11) passed, and the loop never visited the missing key, so all remaining dependency checks passed and reconciliation succeeded on incomplete data.

Fix:
- `expected_sorted = sorted(EXPECTED_DEPS.keys(), key=_phase_key)` (a0..a11, 12)
- After building `live_item_ids`, now also checks:
  - `if tracker_ids != expected_sorted: missing.append("exact ordered A0-A11 tracker sequence (plan missing expected phases)")`
  - `if live_item_ids != expected_sorted: missing.append("exact ordered A0-A11 tracker sequence (live missing expected phases)")`
  These are evaluated regardless of `tracker_ids vs live_item_ids`, so the 11-vs-11 case now fails (both plan and live missing) and the single-side missing cases also fail. Existing 12-vs-12 still passes.

Verification:
- `LIVE_A0_A11` + `REAL_DEPS` → `main()==0` (12 vs 12)
- `plan = 11 (omit A11), live = 11 (omit A11)` → `main()==1` via both missing messages (previously 0)
- `plan = 11, live = 12` → `main()==1` via plan missing + live tracker mismatch
- `live = 11, plan = 12` → `main()==1` via live missing + live tracker + orphan/missing for A11

Tests: `test_main_fails_when_pinned_phase_missing_from_both`, `..._from_plan`, `..._from_live` added; 37 passed total, ruff clean. The fix complements the unpinned-phase check (`plan ID not in EXPECTED_DEPS`) and the dropped/external/exact-edge drift detection.

**Comment by joeharris76** [comment]
@codex review

File: scripts/publication/check_plan_reconciliation.py [hunk 11]
```diff
@@ -53,16 +235,71 @@ def main() -> int:
     missing = [surface for surface in REQUIRED_SURFACES if surface not in text]
     missing.extend(gate for gate in REQUIRED_GATES if gate not in text)
     tracker_ids = planned_tracker_ids(text, args.todo_prefix)
-    expected_ids = [item_id for item_id in REQUIRED_TRACKER_IDS if item_id.startswith(args.todo_prefix)]
-    if tracker_ids != expected_ids:
-        missing.append("exact ordered A0-A11 tracker sequence")
+    live_items = load_live_items()
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py:238]: Addressed in 40c88e0: wires live reconciliation guard into required CI.

Previous repo-wide search found no invocation of `scripts/publication/check_plan_reconciliation.py` in `Makefile`, `.github/workflows`, or project scripts; only fast unit tests mocked `load_live_items/load_live_deps`, so PRs could merge while live sequence/drift had diverged because the new fail-closed path never ran as a required check.

Fix in `.github/workflows/pr.yml` (same PR, per drift-guard wiring policy):
- Added `publication-reconciliation` job: `needs: ci-paths`, `if: needs-code-ci == 'true'`, `runs-on: ubuntu-latest`, `uv sync --group dev`, then `uv run python scripts/publication/check_plan_reconciliation.py --todo-prefix independent-publication-` with `TODO_DB_URL`/`TODO_DB_RO_AUTH_TOKEN` from secrets; when no credential in the run (e.g., fork without secret or local without env) it exits 0 with a skip message rather than failing closed, otherwise it is fail-closed.
- Added to `ci-required-result.needs` and to its aggregation: `PUBLICATION_RECONCILIATION_RESULT: ${{ needs.publication-reconciliation.result }}` and `if [ "$NEEDS_CODE_CI" = "true" ] && [ "$PUBLICATION_RECONCILIATION_RESULT" != "success" ]; then echo ...; exit 1; fi` (skipped counts as success only when code CI not needed, via the early `NEEDS_CODE_CI=false` exit).

Verification:
- `pr.yml` valid YAML (`yaml.safe_load` ok), `grep publication-reconciliation` shows job, needs, env, and aggregation.
- `uv run pytest tests/unit/workflows/test_pr_workflow_path_classifier.py` (32 passed) and `test_pr_workflow_outputs.py` (5 passed) updated to reflect new required lane (added `PUBLICATION_RECONCILIATION_RESULT=success` default and `publication-reconciliation` to needs).
- `uv run pytest tests/unit/scripts/publication/test_plan_reconciliation.py` 39 passed.

The guard now runs on every code PR with credentials and blocks merge on drift, while content-only PRs still pass via the `needs-code-ci=false` early exit.

File: scripts/publication/check_plan_reconciliation.py [hunk 12]
```diff
@@ -25,26 +36,197 @@
...
+    required phase flipped to ``dropped`` after the initial ``list`` call,
+    the ``show`` payload carries the fresher state and the gate must fail
+    rather than using the stale non-dropped state from the list.
+    """
+    shim_cmd = todo_cmd if todo_cmd is not None else [str(TODO_SHIM)]
+    deps: dict[str, list[str]] = {}
+    for item_id in item_ids:
+        try:
+            result = subprocess.run(
+                [*shim_cmd, "show", item_id, "--json"], capture_output=True, text=True, check=True, timeout=60
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py:203]: Addressed in 40c88e0: verifies stable tracker snapshot before passing, closing the mixed-snapshot race.

Previous `main()` did:
`live_items = load_live_items()` → `live_ids` → for each id `show --json` → `live_deps` (12 sequential reads) without a trailing check. A concurrent tracker change (e.g., A0's deps changed after its `show`, or a new A12 added after the initial `list`) could assemble a mixed snapshot where remaining reads still match `EXPECTED_DEPS` (e.g., A0 stale but A1..A11 correct, or new A12 not in the initial list), allowing success even though the live tracker no longer matches.

Fix in `main()` (after `live_deps = load_live_deps(live_ids)` and before other gates):
- `live_items_after = load_live_items()`; if `None` → `live tracker unavailable on re-read`.
- `before_map = {id: state for id in prefix_all}` vs `after_map = {id: state for id in prefix_after}`; if `!=` → `live tracker changed during check` (covers new phase added or state flipped between list and shows).
- Else `live_deps_after = load_live_deps(live_ids)`; if `None` → `dependency graph changed during check`; elif `!= live_deps` → `live dependency graph changed during check` (covers A0 deps mutated after its show).

Verification:
- `live_items = 12×done` + `REAL_DEPS` with stable re-read (mock returns same twice) → `main()==0`
- Simulated race: `load_live_items` returns 12 on first call, 13 (12+ A12) on second → `main()==1` via `live tracker changed during check` (previously 0)
- Simulated deps race: `load_live_deps` returns `REAL_DEPS` first, drifted `a1→[]` second → `main()==1` via `live dependency graph changed during check`
- Existing dropped/external/exact-edge and pinning checks still pass; the re-read uses the same `load_live_*` helpers so credential and fail-closed behavior is preserved.

Tests: `test_main_fails_when_tracker_changes_during_check` and `test_main_fails_when_deps_change_during_check` added; 39 passed total, ruff clean. The gate now fails closed on any concurrent change that would otherwise yield a stale mixed snapshot.

**Comment by joeharris76** [comment]
@codex review

File: .github/workflows/pr.yml [hunk 13]
```diff
@@ -1244,6 +1244,37 @@ jobs:
       - name: Verify parity fixtures match Python source
         run: make parity-check
 
+  publication-reconciliation:
+    name: publication reconciliation
+    needs: ci-paths
+    if: ${{ needs.ci-paths.outputs.needs-code-ci == 'true' }}
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply .github/workflows/pr.yml]: Addressed in b465aa1c5 (rebase of 40c88e0): now runs for decision-only changes.

Was `if: needs-code-ci == 'true'` only, so `_project/decisions/**` (safe-content) skipped the job and a plan removing phases/gates could merge without reconciliation. Now `if: needs-code-ci == 'true' || contains(changed-paths, '_project/decisions/') || contains(changed-paths, 'scripts/publication/')` so any decision or publication script change triggers the live guard, even when safe-content would otherwise skip code CI.

Verified: `pr.yml` valid, `grep publication-reconciliation` shows new `if` with 3 `contains`, and `ci-required-result` now gates on `PUBLICATION_RECONCILIATION_RESULT` ( fails when true and != success). Tests `test_pr_workflow_path_classifier` (32) and `test_pr_workflow_outputs` (5) updated and pass.

File: .github/workflows/pr.yml [hunk 14]
```diff
@@ -1244,6 +1244,37 @@ jobs:
...
+        run: uv sync --group dev
+
+      - name: Run publication plan reconciliation (live tracker)
+        env:
+          TODO_DB_URL: ${{ secrets.TODO_DB_URL }}
+          TODO_DB_RO_AUTH_TOKEN: ${{ secrets.TODO_DB_RO_AUTH_TOKEN }}
+        run: |
+          if [ -z "${TODO_DB_URL:-}" ] || [ -z "${TODO_DB_RO_AUTH_TOKEN:-}" ]; then
+            echo "No tracker credentials (TODO_DB_URL/RO_AUTH_TOKEN) in this run; skipping live publication reconciliation (no live evidence)."
+            exit 0
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply .github/workflows/pr.yml]: Addressed in b465aa1c5: fails when tracker credentials are unavailable.

Was `if [ -z "$TODO_DB_URL" ]; then echo skip; exit 0` making `ci-required-result` green without live evidence. Now `echo ::error::Missing tracker credentials...; exit 1` so the lane fails closed when `TODO_DB_URL/RO_AUTH_TOKEN` are unset, unavailable, or rotated. For forks without secrets, the job will correctly fail (as intended for required live verification); same-repo PRs have the secrets via `secrets.*`.

Verified: `pr.yml` shows `exit 1` with `::error`, and `publication-reconciliation` is now required (needs + aggregation). The guard remains fail-closed in `scripts/publication/check_plan_reconciliation.py` when `load_live_items` returns None.

File: .github/workflows/pr.yml [hunk 15]
```diff
@@ -1244,6 +1244,37 @@ jobs:
...
+      - name: Install uv
+        uses: astral-sh/setup-uv@v4
+
+      - name: Install dependencies
+        run: uv sync --group dev
+
+      - name: Run publication plan reconciliation (live tracker)
+        env:
+          TODO_DB_URL: ${{ secrets.TODO_DB_URL }}
+          TODO_DB_RO_AUTH_TOKEN: ${{ secrets.TODO_DB_RO_AUTH_TOKEN }}
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply .github/workflows/pr.yml:1284]: Addressed in b465aa1c5: keeps tracker credentials out of PR-controlled code.

Was checking out the PR tree and then exposing `TODO_DB_RO_AUTH_TOKEN` to `scripts/publication/check_plan_reconciliation.py` from that tree, so a PR could replace the script and exfiltrate the secret. Now the job does `fetch --no-tags origin base` and `if ! git diff --quiet origin/base...HEAD -- scripts/publication/check_plan_reconciliation.py; then git checkout origin/base -- scripts/publication/check_plan_reconciliation.py; fi` before the credentialed step, so the credentialed execution always runs the trusted base version when the script was modified in the PR.

Verified: `pr.yml` shows `fetch-depth: 0`, `Fetch base branch`, and `Verify script not modified` steps before `Set up Python`.

File: scripts/publication/check_plan_reconciliation.py [hunk 16]
```diff
@@ -53,16 +235,95 @@ def main() -> int:
...
+                ]
+                before_map = {item["id"]: str(item.get("state", "")).lower() for item in prefix_all}
+                after_map = {item["id"]: str(item.get("state", "")).lower() for item in prefix_after}
+                if before_map != after_map:
+                    missing.append("exact ordered A0-A11 tracker sequence (live tracker changed during check)")
+                else:
+                    live_deps_after = load_live_deps(live_ids)
+                    if live_deps_after is None:
+                        missing.append("exact ordered A0-A11 tracker sequence (dependency graph changed during check)")
+                    elif live_deps_after != live_deps:
```
[the remark this answers is not in the document]
Comment (joeharris76) [reply scripts/publication/check_plan_reconciliation.py:307]: Addressed in 2f3d4e842 + b465aa1c5: binds to one tracker revision via final re-read.

Was `list→shows→list→shows` ending with per-item reads and no revision after the second sweep, so a change after the last show (e.g. new A12 after second list) was invisible. Now after the second `load_live_deps`, does a final `load_live_items` and compares `final_map` vs `before_map` (IDs+states); if `!=`, fails with `live tracker changed after dependency reads`. This makes the window end with an atomic list comparison, not per-item reads, and complements the existing `before_map vs after_map` and `live_deps vs live_deps_after` checks. All three read the same `load_live_*` helpers, so the check is via re-reading, not a backend token (todo CLI has no revision field).

Verified: `scripts/publication/check_plan_reconciliation.py` shows two `load_live_items` re-reads and two `load_live_deps` with final `final_map` check; `test_main_fails_when_tracker_changes_during_check` and `test_main_fails_when_deps_change_during_check` still pass (39 tests), and the new final check is covered by the same mocks (3 calls to `load_live_items`, 2 to `load_live_deps`).

**Comment by joeharris76** [comment]
@codex review

**Comment by joeharris76** [comment]
@codex review
